gcloud/_helpers.py

```diff
@@ -48,6 +51,7 @@
     (?P<nanos>\d{1,9})                       # nanoseconds, maybe truncated
     Z                                        # Zulu
 """, re.VERBOSE)
+DEFAULT_CONFIGURATION_PATH = '~/.config/gcloud/configurations/config_default'
```

Comment:
Surprisingly yes!

> On Unix and Windows, return the argument with an initial component of ~ or ~user replaced by that user‘s home directory.

> On Windows, HOME and USERPROFILE will be used if set, otherwise a combination of HOMEPATH and HOMEDRIVE will be used. An initial ~user is handled by stripping the last directory component from the created user path derived above.

https://docs.python.org/2/library/os.path.html#os.path.expanduser

If the file can't be found/read then this check will fail and another one will be attempted.